app/models/template.rb

```diff
@@ -31,6 +35,28 @@ def preview_host_collection
     Host.authorized(:view_hosts).order(:name).limit(100)
   end
 
+  def metadata
+    "<%#\n#{to_export(false).to_yaml.sub(/\A---$/, '').strip}\n%>\n\n"
```

Comment:
Is the second `\n` required here? When exporting a template with metadata, it causes an additional newline between the new metadata segment and whatever follows (which could be a problem in a template containing a shebang line.)